pkg/azuredisk/nodeserver_v2.go

```diff
@@ -90,9 +97,52 @@ func (d *DriverV2) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolume
 		return nil, status.Errorf(codes.Internal, "Failed to find disk on lun %s. %v", lunStr, err)
 	}
 
-	source, err := d.nodeProvisioner.GetDevicePathWithLUN(int(lun))
+	// check if AzVolumeattachment is in Attached state if the disk detachment is not in process
+	if state, getErr := d.crdProvisioner.GetAzVolumeAttachmentState(ctx, diskURI, d.NodeID); getErr != nil {
+		getErr = status.Errorf(codes.Internal, "failed to get current attachment state for volume (%s) and node (%s): %v", diskURI, d.NodeID, getErr)
+		klog.Error(getErr)
+		return nil, getErr
+	} else if !d.nodeProvisioner.IsDetachInProcess(diskURI) {
```

Comment:
Is this logic correct? I am reading "if the disk we get the disk attachment state and no detach is in progress, we error out"; seems like the normal case here.